lms/djangoapps/discussion_api/tests/test_views.py

```diff
@@ -707,6 +707,49 @@ def test_profile_image_requested_field(self):
             response_users = response_thread['users']
             self.assertEqual(expected_profile_data, response_users[response_thread['author']])
 
+    def test_profile_image_requested_field_anonymous_user(self):
+        """
+        Tests profile_image in requested_fields for thread created with anonymous user
+        """
+        source_threads = [
+            {
+                "type": "thread",
+                "id": "test_thread",
+                "course_id": unicode(self.course.id),
+                "commentable_id": "test_topic",
+                "group_id": None,
+                "user_id": None,
+                "username": None,
+                "anonymous": True,
+                "anonymous_to_peers": True,
+                "created_at": "2015-04-28T00:00:00Z",
```

Comment:
Most of this is the same for each test thread.  Can we create a help function to get rid of most of this code and focus in on the parts that are important?